### PR TITLE
feat: simplify block to only be data and cid

### PR DESCRIPTION
**Pull request by dignifiedquire**
Ref https://github.com/ipfs/js-ipfs-repo/pull/123

### CODE REVIEW

**Comment by dignifiedquire** [comment]
ref https://github.com/ipfs/js-ipfs/issues/787

**Comment by daviddias** [comment]
@dignifiedquire do I need to say it again? no yarn.lock :(

File: src/index.js [hunk 1]
```diff
@@ -1,90 +1,56 @@
...
-      }
-      this._cache[hashFunc] = multihash
-      callback(null, multihash)
-    })
+   * Check if the given value is a Block.
+   *
+   * @param {any} other
+   * @returns {bool}
+   */
+  static isBlock (other) {
```
Comment (daviddias) [inline src/index.js:61]: Still making my mind over this, but currently, I strongly feel that public module methods shouldn't be encapsulated in Class objects and made public with `static`.

I've seen a lot of cases of people starting to use `static` and then forgetting what it really means and start mutating the state.

Plus, it is way more understandable from a Node.js/CJS user if the "module.exports" is the only place that tells me what gets exported.

File: src/index.js [hunk 2]
```diff
@@ -1,90 +1,56 @@
...
+ * const block = new Block(new Buffer('a012d83b20f9371...'))
  */
-function Block (data) {
-  if (!(this instanceof Block)) {
-    return new Block(data)
-  }
+class Block {
+  constructor (data, cid) {
+    if (!data || !Buffer.isBuffer(data)) {
+      throw new Error('data must be a buffer')
```
Comment (daviddias) [inline src/index.js]: first argument must be a Buffer

File: src/index.js [hunk 3]
```diff
@@ -1,90 +1,56 @@
...
+    }
 
-  if (!data) {
-    throw new Error('Block must be constructed with data')
-  }
+    if (!cid || !CID.isCID(cid)) {
+      throw new Error('cid must be a CID')
```
Comment (daviddias) [inline src/index.js]: second argument must be a CID

**Review by daviddias — CHANGES_REQUESTED** [review]
- [x] remove yarn.lock
- [x] update to aegir 11 (I'm being fair 😇)

File: src/index.js (hunk 1 above)
Comment (dignifiedquire) [reply src/index.js:61]: Declaring `Block.isBlock = ...` doesn't seem to be any better to me in terms of understanding.

Comment (dignifiedquire) [reply src/index.js:61]: also we started using this a while ago in modules like cid

Comment (daviddias) [reply src/index.js:61]: where? https://github.com/ipld/js-cid/blob/master/src/index.js#L133-L135

Comment (dignifiedquire) [reply src/index.js:61]: fine

Comment (dignifiedquire) [reply src/index.js:61]: one bonus is that jsdoc picks up the right docs for this if we write it as static

**Comment by dignifiedquire** [comment]
applied all cr except the static thing

File: src/index.js (hunk 1 above)
Comment (daviddias) [reply src/index.js:61]: hm.....

**Pull request merged**